crates/revm/revm-inspectors/src/tracing/builder/geth.rs

```diff
@@ -0,0 +1,104 @@
+//! Geth trace builder
+
+use crate::tracing::{types::CallTraceNode, TraceInspectorConfig};
+use reth_primitives::{Address, JsonU256, H256, U256};
+use reth_rpc_types::trace::geth::*;
+use revm::interpreter::opcode;
+use std::collections::{BTreeMap, HashMap};
+
+/// A type for creating geth style traces
+#[derive(Clone, Debug)]
+pub struct GethTraceBuilder {
+    /// Recorded trace nodes.
+    nodes: Vec<CallTraceNode>,
+    /// How the traces were recorded
+    _config: TraceInspectorConfig,
+}
+
+impl GethTraceBuilder {
+    /// Returns a new instance of the builder
+    pub(crate) fn new(nodes: Vec<CallTraceNode>, _config: TraceInspectorConfig) -> Self {
+        Self { nodes, _config }
+    }
+
+    /// Recursively fill in the geth trace by going through the traces
+    ///
+    /// TODO rewrite this iteratively
+    fn add_to_geth_trace(
+        &self,
+        storage: &mut HashMap<Address, BTreeMap<H256, H256>>,
+        trace_node: &CallTraceNode,
+        struct_logs: &mut Vec<StructLog>,
+        opts: &GethDebugTracingOptions,
+    ) {
+        let mut child_id = 0;
+        // Iterate over the steps inside the given trace
+        for step in trace_node.trace.steps.iter() {
+            let mut log: StructLog = step.into();
+
+            // Fill in memory and storage depending on the options
+            if !opts.disable_storage.unwrap_or_default() {
+                let contract_storage = storage.entry(step.contract).or_default();
+                if let Some((key, value)) = step.state_diff {
+                    contract_storage.insert(key.into(), value.into());
+                    log.storage = Some(contract_storage.clone());
+                }
+            }
+            if opts.disable_stack.unwrap_or_default() {
+                log.stack = None;
+            }
+            if !opts.enable_memory.unwrap_or_default() {
+                log.memory = None;
+            }
```

Comment:
by default, the stack is enabled and memory is disabled. that's the intention, right?

Reply:
depends on the target, for parity style we don't need any of them.